bundles/org.openhab.binding.deconz/src/main/resources/ESH-INF/thing/sensor-thing-types.xml

```diff
@@ -475,5 +475,53 @@
 		<state readOnly="true"/>
 	</channel-type>
 
+	<thing-type id="thermostat">
+		<supported-bridge-type-refs>
+			<bridge-type-ref id="deconz"/>
+		</supported-bridge-type-refs>
+		<label>Thermostat</label>
+		<description>A Thermostat sensor/actor</description>
+		<channels>
+			<channel typeId="temperature" id="temperature"/>
+			<channel typeId="heatsetpoint" id="heatsetpoint"/>
```

Comment:
If the channel is added from the thing-type, you don't need to add it dynamically, also below. If the channels are mandatory, adding them here is the better choice (simplifies code).

Reply:
Agree.